src/net.cpp

```diff
@@ -539,7 +539,7 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes)
     {
         // get current incomplete message, or create a new one
         if (vRecvMsg.empty() || vRecvMsg.back().complete())
-            vRecvMsg.push_back(CNetMessage(Params().MessageStart(), SER_NETWORK, nRecvVersion));
+            vRecvMsg.push_back(CNetMessage(GetMagic(Params()), SER_NETWORK, nRecvVersion));
 
```

Comment:
i see what you're doing here but i don't see how this prevents us from sending an initial VERSION with the legacy magic before receiving and setting the cashmagic value on receiving a cash VERSION message...won't we still get a disconnect when ABC nodes stop processing legacy transactions and we send them the legacy VERSION message first?

Reply:
you are right indeed. 

to avoid sending our first `version` cmd using the legacy magic we should change `fUsesCashMagic`default from false to true, but this is something I need to to only for the cash fork, hence this is why I need to add another commit to change the default in case of `#ifdef BITCOIN_CASH`

Reply:
Actually I looked at the raw data captured on the wire and even w/o changing the default value of `fUsesCashMagic` the entire connection will use the new magic.  

Looking at the code now, it seems that for some reason with this code modifications the other cash peer always send `version` message first.